### ISSUE TITLE
pytest-tap unable to unregister terminalreporter?

### ISSUE THREAD

**Issue by gregsadetsky** [python-tap/pytest-tap#24]
Hi!

We're using `pytest==3.1.0` and `pytest-tap==2.0` and are seeing the following output:

```
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/main.py", line 101, in wrap_session
    config._do_configure()
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/config.py", line 906, in _do_configure
    self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 750, in call_historic
    self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "/app/.heroku/python/lib/python3.6/site-packages/pytest_tap/plugin.py", line 45, in pytest_configure
    config.pluginmanager.unregister(reporter)
  File "/app/.heroku/python/lib/python3.6/site-packages/_pytest/vendored_packages/pluggy.py", line 393, in unregister
    assert plugin is not None, "one of name or plugin needs to be specified"
AssertionError: one of name or plugin needs to be specified
```

Has anything changed in the latest versions of pytest which would break pytest-tap? Is this another kind of incompatibility?

Thanks in advance!

Greg

**Comment by mblayman** [comment]
Hi, @gregsadetsky. Thanks for the issue report. I haven't tested on Python 3.6 and the latest Travis CI build was tested against pytest version 2.9.2 (https://travis-ci.org/python-tap/pytest-tap/jobs/148770254).

Since you're running pytest version 3.1.0, it seems likely to me that the major version bump in pytest created some incompatibility. If you'd like to update `tox.ini` and `.travis.yml` to test Python 3.6, I'm guessing a pull request would quickly reveal if this was a common problem.

I'd happily merge and release a new version to make sure that `pytest-tap` stays compatible with the latest pytest. Thanks!

**Comment by gregsadetsky** [comment]
Just did, thanks! (+ updated requirements.txt) Looking forward to the Jenkins test

**Comment by mblayman** [comment]
It looks like we have a mystery on our hands. The build seems to have executed with the latest Python and pytest without any trouble. I'm not certain what could be different about your environment.

I'll try running a quick test locally to see what happens for me.

**Comment by mblayman** [comment]
I can't recreate the problem you encountered in my local development environment.

```bash
➜  projects mkdir tap36
➜  projects cd tap36
➜  tap36 python3 -m venv venv
➜  tap36 . venv/bin/activate
(venv) ➜  tap36 pip install pytest pytest-tap
Collecting pytest
  Downloading pytest-3.1.0-py2.py3-none-any.whl (178kB)
    100% |████████████████████████████████| 184kB 4.1MB/s 
Collecting pytest-tap
  Downloading pytest_tap-2.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./venv/lib/python3.6/site-packages (from pytest)
Collecting py>=1.4.29 (from pytest)
  Using cached py-1.4.33-py2.py3-none-any.whl
Collecting tap.py (from pytest-tap)
  Downloading tap.py-2.1-py2.py3-none-any.whl (64kB)
    100% |████████████████████████████████| 71kB 11.3MB/s 
Installing collected packages: py, pytest, tap.py, pytest-tap
Successfully installed py-1.4.33 pytest-3.1.0 pytest-tap-2.0 tap.py-2.1
(venv) ➜  tap36 vim test_foo.py
(venv) ➜  tap36 cat test_foo.py 
def test_foo():
    assert 42 == 42
(venv) ➜  tap36 pytest --tap-stream
ok 1 - test_foo
1..1
```

Can you describe the context that this is running in?

**Comment by mblayman** [comment]
Since I'm unable to reproduce this and there hasn't been much feedback after the initial report, I'm going to assume that this is all resolved. Please feel free to open another issue if you have more to report and this is still a problem. Thanks for the contribution!

**Comment by gregsadetsky** [comment]
Hey Mark,

My apologies for the silence! Our CI setup (within which we attempted to use TAP) snowballed into a whole headache project :-) We just got it under control a few days ago, and should be able to look into TAP again. I will report back if things aren't working, and will make an attempt at creating a minimally reproducible example.

Until then, sorry again & thanks for your great work and for the follow up!

All the best,

Greg

**Comment by mblayman** [comment]
No problem, Greg. I hope everything works out for you.

**Comment by archfear** [comment]
This issue happens when the tests are run in parallel with `pytest-xdist` and `--tap-stream` is used. I've made a pull request with a fix here: #27

**Comment by mblayman** [comment]
@gregsadetsky, @archfear: I released an updated version of the plugin with the `pytest-xdist` fix. The new version is 2.1. I hope that helps in your environments!
